persist/sqldb/workflow_archive.go

```diff
@@ -72,6 +73,10 @@ func (r *workflowArchive) IsEnabled() bool {
 	return true
 }
 
+func (r *workflowArchive) ValidateWorkflow(wf *wfv1.Workflow) error {
```

Comment:
this responsibily does not fit within the archive - what does validation to do with archiving?